packages/parser/src/fs.ts

```diff
@@ -34,6 +34,27 @@ export async function load(filepath: string, themeMeta?: SlidevThemeMeta, conten
     const baseSlide = data.slides[iSlide]
     if (!baseSlide.frontmatter.src) {
       iSlide++
+
+      baseSlide.content = baseSlide.content.replaceAll(
+        /^```(\w+?)\s*\[([\s\S]+?)\]([\s\S]*?)\n+^```/mg,
+        (full, lang = '', external: string, rest: string) => {
+          const [externalPath, externalRangeStr] = external.split(':').map(i => i.trim())
+          const sourcePath = resolve(dir, 'snippets', externalPath)
```

Comment:
I am not sure should the snippets be put in `${dir}/snippets/${snippetName}` or somewhere else, or should use `userRoot` instead of `dir`.

Reply:
I think the snippet path should be resolved relative to the current slide markdown file. This seems more make sense?

Reply:
I had a look at L76-82 of this file, maybe this is a better choice?

```ts
let sourcePath
if (external.startsWith('/'))
  sourcePath = resolve(dir, external.substring(1))
else if (baseSlide.source?.filepath)
  sourcePath = resolve(dirname(baseSlide.source.filepath), external)
else
  sourcePath = resolve(dir, external)
```

(The same rule as the `src` in frontmatter)